### PR TITLE
Milestone #2 - Create data models and methods

### PR DESCRIPTION
**Pull request by NicolasAndrehh**
# Milestone - Data Models
This pull request includes the following changes:
- Create the following models
    - User
    - Post
    - Comment
    - Like
- Add seeder
- Add `recent_posts` method to User
- Add `update_post_counter` and `recent_comments` methods to Post
- Add `update_comment_counter` method to Comment
- Add `update_like_counter` method to Like

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  belongs_to :user, optional: true
```
Comment (codecaiine) [inline app/models/post.rb:2]: - [ ] [OPTIONAL] I would like to recommend you use `after_save` to call the `#update_post_counter` method every time a new post has been saved to database, then make the `update_post_counter` method private 👍

File: app/models/comment.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+require_relative 'application_record'
+
```
Comment (codecaiine) [inline app/models/comment.rb:2]: - [ ] [OPTIONAL] I would like to recommend you use `after_save :update_comments_counter` 👍

File: app/models/like.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+class Like < ApplicationRecord
+  belongs_to :user, optional: true
```
Comment (codecaiine) [inline app/models/like.rb:2]: - [ ] [OPTIONAL] I would like to recommend you use` after_save :update_likes_counter` 👍

**Review by codecaiine — APPROVED** [review]
Hi @NicolasAndrehh , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- processing data in models works as expected ✔️ 
- associations between models are set up correctly ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me (@codecaiine) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by NicolasAndrehh** [comment]
Hey @codecaiine, thank you so much for the suggestions 😄.

**Pull request merged**